### PR TITLE
created delete product endpoint

### PR DESCRIPTION
**Pull request by ShbairAli**

### CODE REVIEW

File: src/controllers/ProductController.ts [hunk 1]
```diff
@@ -61,4 +78,21 @@ export default class ProductController {
...
+
+  static deleteProduct = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) => {
+    try {
+      const { id } = req.params;
+      await ProductQuery.deleteProduct(Number(id));
+      res.json({
```
Comment (AhmedSafi97) [inline src/controllers/ProductController.ts:90]: `res.sendStatus(204)`

File: src/controllers/ProductController.ts [hunk 2]
```diff
@@ -43,15 +43,32 @@ export default class ProductController {
...
-        `select p.id,p.price,p.title,p.icon,p.description,coalesce(((select SUM(t.quantity) from "TransactionProducts" as t join "Transactions" as ts on ts.id = t."TransactionId" where t.status = 'closed' and ts.type = 'purchase' and t."ProductId" = p.id)-(select
-            SUM(t.quantity)
-          from
-            "TransactionProducts" as t
-            join "Transactions" as ts on ts.id = t."TransactionId"
-          where
-            t.status = 'closed'
-            and ts.type = 'sale'
-            and t."ProductId" = p.id)),0) as "inStock" from "Products" as p;`,
+        `select p.id,
```
Comment (AhmedSafi97) [inline src/controllers/ProductController.ts:46]: changes outside of your task scope!

File: src/controllers/ProductController.ts (hunk 1 above)
Comment (AhmedSafi97) [inline src/controllers/ProductController.ts:90]: `res.sendStatus(204)`

**Review by AhmedSafi97 — CHANGES_REQUESTED** [review]

File: src/controllers/ProductController.ts (hunk 1 above)
Comment (ShbairAli) [reply src/controllers/ProductController.ts:90]: done 👍

**Review by AhmedSafi97 — APPROVED** [review]

**Pull request merged**